akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/internal/ReplicationProjectionProviderAdapter.scala

```diff
@@ -37,12 +39,14 @@ private[akka] object ReplicationProjectionProviderAdapter {
           throw new IllegalArgumentException(
             s"The source provider is required to implement akka.projection.BySlicesSourceProvider but ${noSlices.getClass} does not")
       }
+
+      val sourceProviderAdapter = providerWithSlices match {
+        case delegate: CanTriggerReplay => new ScalaBySlicesSourceProviderAdapterWithCanTriggerReplay(delegate)
+        case _                          => new ScalaBySlicesSourceProviderAdapter(providerWithSlices)
```

Comment:
I'm confused. In the previously failing test, this was not the adapter that was used in the end, it was the r2dbc adapter.

Reply:
`akka.projection.internal.ScalaBySlicesSourceProviderAdapter` adapts a Scala provider into Java DSL while `akka.projection.r2dbc.internal.BySliceSourceProviderAdapter` adapts a Java provider to Scala DSL.

Still feels like both should be in projection core though? (And possibly detect the other one and unwrap in case we do adapt an already adapted source)

Reply:
Done in [90430a6](https://github.com/akka/akka-projection/pull/1094/commits/90430a65825baaed26e4a19d994185db6c5171bd)